src/energyplus/Test/HeatExchangerDesiccantBalancedFlowPerformanceDataType1_GTest.cpp

```diff
@@ -0,0 +1,277 @@
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+#include "EnergyPlusFixture.hpp"
+
+#include "../ForwardTranslator.hpp"
+
+#include "../../model/Model.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlow.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlow_Impl.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlowPerformanceDataType1.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlowPerformanceDataType1_Impl.hpp"
+
+#include "../../model/AirLoopHVAC.hpp"
+#include "../../model/AirLoopHVAC_Impl.hpp"
+#include "../../model/Node.hpp"
+#include "../../model/Node_Impl.hpp"
+#include "../../model/AirLoopHVACOutdoorAirSystem.hpp"
+#include "../../model/AirLoopHVACOutdoorAirSystem_Impl.hpp"
+#include "../../model/HVACTemplates.hpp"
+
+#include <utilities/idd/HeatExchanger_Desiccant_BalancedFlow_PerformanceDataType1_FieldEnums.hxx>
+#include <utilities/idd/IddEnums.hxx>
+
+using namespace openstudio::energyplus;
+using namespace openstudio::model;
+using namespace openstudio;
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_HeatExchangerDesiccantBalancedFlowPerformanceDataType1) {
+  Model m;
+
+  AirLoopHVAC loop = addSystemType3(m).cast<AirLoopHVAC>();
+
+  AirLoopHVACOutdoorAirSystem oaSystem = loop.airLoopHVACOutdoorAirSystem().get();
+
+  Node oaNode = oaSystem.outboardOANode().get();
+
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p(m);
+  HeatExchangerDesiccantBalancedFlow hx(m, p);
+  EXPECT_TRUE(hx.addToNode(oaNode));
+
+  ForwardTranslator ft;
+  Workspace w = ft.translateModel(m);
+
+  WorkspaceObjectVector idfHXPs(w.getObjectsByType(IddObjectType::HeatExchanger_Desiccant_BalancedFlow_PerformanceDataType1));
+  ASSERT_EQ(1u, idfHXPs.size());
+  WorkspaceObject idfHXP(idfHXPs[0]);
+
+  EXPECT_EQ("Autosize", idfHXP.getString(HeatExchanger_Desiccant_BalancedFlow_PerformanceDataType1Fields::NominalAirFlowRate, false).get());
+  EXPECT_EQ("Autosize", idfHXP.getString(HeatExchanger_Desiccant_BalancedFlow_PerformanceDataType1Fields::NominalAirFaceVelocity, false).get());
+  EXPECT_EQ(0.0, idfHXP.getDouble(HeatExchanger_Desiccant_BalancedFlow_PerformanceDataType1Fields::NominalElectricPower, false).get());
+  EXPECT_EQ(-7.18302E+00,
```

Comment:
You don't necessarily have to change anything in this case (up to you), but I personally like to make my FT tests as explicit as possible and ensure I catch all corner cases, for the following reasons:

* Here you're relying on defaults, which makes it less clear this is actually accurate when you test the IDF (you don't see those default values that are in the Ctor in a file "far away")
* if we change the model default, the FT test needs to be adjusted
* By having multiple fields autosized here, you wouldn't catch a copy paste error if for eg we were incorrectly writing the modelObject.nominalAirFlowRate() to the E+ NominalAirFaceVelocity field.